katharsis-core/src/main/java/io/katharsis/resource/include/IncludeLookupSetter.java

```diff
@@ -1,167 +1,218 @@
+	 * 
+	 * @param resources
+	 * @param relationshipField
+	 * @param queryAdapter
+	 * @param parameterProvider
+	 * @return list of all loaded properties. Collections are flattened into a single collection.
+	 */
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	private Set loadRelationships(ResourceInformation resourceInformation, List resources, ResourceField relationshipField,
+			QueryAdapter queryAdapter, RepositoryMethodParameterProvider parameterProvider) {
+		RegistryEntry rootEntry = resourceRegistry.getEntry(resourceInformation.getResourceType());
+
+		List<Serializable> resourceIds = getIds(resources, resourceInformation);
+
+		boolean isMany = Iterable.class.isAssignableFrom(relationshipField.getType());
+		boolean isList = List.class.isAssignableFrom(relationshipField.getType());
+		Class<?> relationshipFieldClass = relationshipField.getElementType();
+
+		Set loadedEntities = new HashSet();
+
+		RelationshipRepositoryAdapter relationshipRepository = rootEntry.getRelationshipRepositoryForClass(relationshipFieldClass,
+				parameterProvider);
+		if (relationshipRepository != null) {
+			Map<Object, JsonApiResponse> responseMap;
+			if (isMany) {
+				responseMap = relationshipRepository.findBulkManyTargets(resourceIds, relationshipField.getUnderlyingName(),
+						queryAdapter);
+			}
+			else {
+				responseMap = relationshipRepository.findBulkOneTargets(resourceIds, relationshipField.getUnderlyingName(),
+						queryAdapter);
+			}
+
+			for (Object resource : resources) {
+				ResourceField rootIdField = resourceInformation.getIdField();
+				Serializable id = (Serializable) PropertyUtils.getProperty(resource, rootIdField.getUnderlyingName());
+				JsonApiResponse response = responseMap.get(id);
+				if (response != null) {
+					// set the relation
+					Object entity = response.getEntity();
+					PropertyUtils.setProperty(resource, relationshipField.getUnderlyingName(), entity);
+
+					addAll(loadedEntities, entity);
+				}else{
+					// null the relation
+					PropertyUtils.setProperty(resource, relationshipField.getUnderlyingName(), null);
+				}
+			}
+		}
+
+		return loadedEntities;
+	}
+
+	private void addAll(Set<Object> set, Object entity) {
+		if (entity instanceof Iterable) {
+			Iterator<?> iterator = ((Iterable<?>) entity).iterator();
+			while (iterator.hasNext()) {
+				set.add(iterator.next());
+			}
+		}
+		else {
+			set.add(entity);
+		}
+	}
+
+	@SuppressWarnings("rawtypes")
+	private List<Serializable> getIds(List resources, ResourceInformation resourceInformation) {
+		ResourceField rootIdField = resourceInformation.getIdField();
+		List<Serializable> ids = new ArrayList<>();
+		for (Object resource : resources) {
+			Serializable id = (Serializable) PropertyUtils.getProperty(resource, rootIdField.getUnderlyingName());
+			ids.add(id);
+		}
+		return ids;
+	}
+
+	private Class<?> getClassFromField(ResourceField relationshipField) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getClassFromField" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)